backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/Document.java

```diff
@@ -1354,26 +1356,39 @@ private void updateReadonlyAndPropagate(final ReasonSupplier reason)
 	}
 
 	private DocumentReadonly computeReadonly()
+	{
+		final boolean fieldsReadOnlyBasedOnDecorators = getEntityDescriptor().getDocumentDecorators() != null
+				&& getEntityDescriptor().getDocumentDecorators()
+				.stream()
+				.anyMatch(documentDecorator -> documentDecorator.isReadOnly(this));
+
+		final boolean fieldsReadOnly = fieldsReadOnlyBasedOnDecorators || computeDefaultFieldsReadOnly().booleanValue();
+
+		return DocumentReadonly.builder()
+				.parentActive(parentReadonly.isActive()).active(isActive())
+				.processed(parentReadonly.isProcessed() || isProcessed())
+				.processing(parentReadonly.isProcessing() || isProcessing())
+				.fieldsReadonly(fieldsReadOnly)
+				.build();
```

Comment:
consider improving performances:
1. if `computeDefaultFieldsReadOnly().booleanValue()` is true then `fieldsReadOnlyBasedOnDecorators` is not relevant so you don't have to compute it.
2. check the implementation of `DocumentReadonly.computeFieldReadonly`. Remark that there are some cases when `fieldsReadonly` is not relevant. Consider introducing DocumentReadonly.fieldsReadOnlyBasedOnDecorators which could be some MemoizedSupplier so the computation is deferred and also cached.